cumulus_library/apis/umls.py

```diff
@@ -0,0 +1,152 @@
+        :keyword validator_key: A UMLS API key for the calling application. Will be
+            set to the value of api_key if None (which is the current expected
+            behavior, since we don't want to be distributing UMLS keys)
+        """
+
+        if api_key is None:
+            api_key = os.environ.get("UMLS_API_KEY", None)
+            if api_key is None:
+                raise errors.ApiError("No UMLS API key provided")
+        self.api_key = api_key
+        if validator_key is None:
+            validator_key = api_key
+        self.validator_key = validator_key
+
+        auth_payload = {"validatorApiKey": self.validator_key, "apiKey": self.api_key}
+        self.session = requests.Session()
+        response = self.session.get(
+            "https://utslogin.nlm.nih.gov/validateUser", params=auth_payload
+        )
+        if response.status_code == 401:
+            raise errors.ApiError("Invalid UMLS API validator key")
+        if response.text != "true":
+            raise errors.ApiError("Invalid UMLS API key")
+        self.session.auth = requests.auth.HTTPBasicAuth("apikey", api_key)
+
+    def get_vsac_valuesets(
+        self, url: str | None = None, oid: str | None = None
+    ) -> list[dict]:
+        """Gets a valueset, and any nested valuesets, from the VSAC API
+
+        :keyword url: an URL to target for a valueset (typically expected)
+        :keyword oid: A valuset OID
+        :returns: A list, containing the valueset and any referenced
+            valuesets.
+
+        Documentation on this API is available at
+        https://www.nlm.nih.gov/vsac/support/usingvsac/vsacfhirapi.html
+
+
+        TODO: do we need to support the FHIR operators?
+        TODO: do we need to support the v2 API?
+        https://www.nlm.nih.gov/vsac/support/usingvsac/vsacsvsapiv2.html
+        """
+        if url is None:
+            url = "https://cts.nlm.nih.gov/fhir/res/ValueSet"
+        if oid:
+            url = f"{url}/{oid}"
+
+        # If we're inspecting url references in a VSAC response, they come back
+        # specifying a url that does not align with the actual implemented rest
+        # APIs, so we do some massaging
+        if "http:" in url:
+            url = url.replace("http:", "https:")
+        if "/res/" not in url:
+            url = url.replace("/fhir/", "/fhir/res/")
+        response = self.session.get(url)
+        if response.status_code == 404:
+            raise errors.ApiError(f"Url not found: {url}")
+        res_json = response.json()
+        if "compose" in res_json and "include" in res_json["compose"]:
+            nested_valuesets = []
+            for record in res_json["compose"]["include"]:
+                if "valueSet" in record:
+                    nested_get = self.get_vsac_valuesets(url=record["valueSet"][0])
+                    nested_valuesets.append(nested_get[0])
+
+            return [response.json(), *nested_valuesets]
+        return response.json()
```

Comment:
I think you want `[response.json()]` here?

Or just skip the forking logic and do something like:
```python
all_valuesets = [response.json()]
records = res_json.get("compose", {}).get("include", [])
for record in records:
    ...
    all_valuesets.append()
return all_valuesets
```

Reply:
i had [response.json()] here at one point and was getting an extra layer of list at some point. lemme go back and check.